src/pages/UserProfilePage/UserProfilePage.tsx

```diff
@@ -0,0 +1,73 @@
+import { UserCircleIcon } from '@heroicons/react/24/solid';
+import classes from './UserProfilePage.module.scss';
+import { getUser } from 'src/services/UserService';
+import Layout from 'src/components/Layout/Layout';
+import { useEffect, useState } from 'react';
+import User from 'src/models/User';
+import { useParams } from 'react-router-dom';
+
+const UserProfilePage: React.FC = () => {
+  const [user, setUser] = useState<User>();
+  const { id } = useParams();
+
+  useEffect(() => {
+    getUser(Number.parseInt(id ?? ''))
+      .then((response) => {
+        if (response.status == 200) {
+          setUser(response.data);
+        }
+      })
+      .catch((error) => {
+        if (error.response.status == 404) {
+          setUser({
+            email: '',
+            firstName: '',
+            lastName: '',
+            username: 'User not found.',
+            phoneNumber: '',
+            postsCount: 0,
+            friendsCount: 0
+          });
+        }
```

Comment:
If user doesn't exist this page shouldn't be accessible, please just do redirect here